tests/virtualization/universal/guest_management.pm

```diff
@@ -19,17 +19,6 @@ use testapi;
 use utils;
 
 sub run {
-    ## Ensure every guest has <on_reboot>restart</on_reboot>
-    foreach my $guest (keys %virt_autotest::common::guests) {
-        my $filename = "$guest.xml";
-        assert_script_run("virsh dumpxml $guest > $filename");
-        if (script_run("! cat $filename | grep 'on_reboot' | grep -v 'restart'") != 0) {
-            record_info("$guest bsc#1153028", "Setting on_reboot=restart for $guest");
-            assert_script_run("sed -i '/\<on_reboot\>/c\  <on_reboot\>restart\</on_reboot\>/' $filename");
-            assert_script_run("virsh define $filename");
-        }
-        script_run("rm $filename");
-    }
```

Comment:
Why do you remove this?

Reply:
I moved this part to `prepare_guests`.

In `guest_management` it was too late to apply those changes. Changing the reboot behaviour does not alter the state of the currently running VM and needs to be done before starting the virtual machine.
So this did fix bsc#1153028, but the reboot still failed, because the changes in the `xml` definition get applied after the NEXT reboot.